repo/format/upgrade_lock.go

```diff
@@ -86,6 +103,11 @@ func (m *Manager) CommitUpgrade(ctx context.Context) error {
 		return errors.New("no upgrade in progress")
 	}
 
+	// poison V0 index so that old readers won't be able to open it.
+	if err := WriteLegacyIndexPoisonBlob(ctx, m.blobs); err != nil {
+		log(ctx).Errorf("unable to write legacy index poison blob: %v", err)
+	}
+
 	// restore the old format version
```

Comment:
Can upgrade succeed without `WriteLegacyIndexPoisonBlob()` succeeding ? What would be the implications for unlocking the repo when that happens ?

Reply:
Never mind, this was also the existing behavior. So, you are not changing it.